app/jobs/get_next_ai_message_job.rb

```diff
@@ -33,7 +33,7 @@ def perform(user_id, message_id, assistant_id, attempt = 1)
 
     response = Current.set(user: @user, message: @message) do
       ai_backend.new(@conversation.user, @assistant, @conversation, @message)
-      .stream_next_conversation_message do |content_chunk|
+      .stream_next_conversation_message.each do |content_chunk|
```

Comment:
This is a bug. “stream_next_conversation_message” is not an array, it’s a method call. We’re calling:

`ai_backend.new.stream_next_conversation_message` and this method call takes a block so we’re passing in `do … end` to it

Reply:
Ahh ok now I think I understand this part! I need to write the method  stream_handler for gemini respond @krschacht thanks!